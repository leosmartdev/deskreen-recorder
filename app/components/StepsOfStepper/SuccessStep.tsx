/* eslint-disable react/destructuring-assignment */
import React, { useCallback, useEffect } from 'react';
import { Button, H5, Icon, Text } from '@blueprintjs/core';
import { Row, Col } from 'react-flexbox-grid';
import { useTranslation } from 'react-i18next';

interface SuccessStepProps {
  handleReset: () => void;
}

const SuccessStep: React.FC<SuccessStepProps> = (props: SuccessStepProps) => {
  const { t } = useTranslation();

  useEffect(() => {
    document
      .querySelector('#top-panel-connected-devices-list-button')
      ?.classList.remove('pulse-not-infinite');

    document
      .querySelector('#top-panel-connected-devices-list-button')
      ?.classList.add('pulse-not-infinite');

    setTimeout(() => {
      document
        .querySelector('#top-panel-connected-devices-list-button')
        ?.classList.remove('pulse-not-infinite');
    }, 4000);
  }, []);

  const handleTextConnectedListMouseEnter = useCallback(() => {
    document
      .querySelector('#top-panel-connected-devices-list-button')
      ?.classList.add('pulsing');
  }, []);

  const handleTextConnectedListMouseLeave = useCallback(() => {
    document
      .querySelector('#top-panel-connected-devices-list-button')
      ?.classList.remove('pulsing');
  }, []);

  return (
    <Col
      xs={8}
      md={6}
      style={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
      }}
    >
      <Row center="xs">
        <Col xs={12}>
          <Icon icon="endorsed" iconSize={35} color="#0F9960" />
          <H5>{t('Done!')}</H5>
        </Col>
      </Row>
      <Row center="xs">
        <Col xs={10}>
          <div style={{ marginBottom: '10px' }}>
            <Text>{t('Now you can see your screen on other device')}</Text>
          </div>
          <div
            id="connected-devices-list-text-success"
            onMouseEnter={handleTextConnectedListMouseEnter}
            onMouseLeave={handleTextConnectedListMouseLeave}
            style={{
              marginBottom: '25px',
              textDecoration: 'underline dotted',
            }}
          >
            <Text className="">
              {t(
                'You can manage connected devices by clicking Connected Devices button in top panel'
              )}
            </Text>
          </div>
        </Col>
      </Row>
      <Button
        intent="primary"
        onClick={props.handleReset}
        icon="repeat"
        style={{ borderRadius: '100px' }}
      >
        {t('Connect New Device')}
      </Button>
    </Col>
  );
};

export default SuccessStep;
