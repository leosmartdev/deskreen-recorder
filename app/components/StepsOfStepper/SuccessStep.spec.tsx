import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import SuccessStep from './SuccessStep';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

const connectNewDeviceButtonSelector = 'button.bp3-button.bp3-intent-primary';

function setup(handleResetMock = () => {}) {
  const getWrapper = () =>
    mount(
      <>
        <Router>
          <SuccessStep handleReset={handleResetMock} />
        </Router>
      </>
    );

  const component = getWrapper();

  const buttons = {
    connectNewDeviceButton: component.find(connectNewDeviceButtonSelector),
  };

  return {
    component,
    buttons,
  };
}

it('should match exact snapshot', () => {
  const { component } = setup();
  expect(EnzymeToJson(component)).toMatchSnapshot();
});

describe('when "Connect New Device" button is clicked', () => {
  it('should call handleReset injected functional property', () => {
    const mockHandleResetCallback = jest.fn();
    const { buttons } = setup(mockHandleResetCallback);

    buttons.connectNewDeviceButton.simulate('click');

    expect(mockHandleResetCallback).toBeCalled();
  });
});
