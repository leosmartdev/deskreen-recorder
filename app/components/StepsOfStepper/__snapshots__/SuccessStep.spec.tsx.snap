// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SuccessStep
      handleReset={[Function]}
    >
      <Col
        md={6}
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
        xs={8}
      >
        <div
          className="col-xs-8 col-md-6"
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Row
            center="xs"
          >
            <div
              className="row center-xs"
            >
              <Col
                xs={12}
              >
                <div
                  className="col-xs-12"
                >
                  <Blueprint3.Icon
                    color="#0F9960"
                    icon="endorsed"
                    iconSize={35}
                  >
                    <span
                      className="bp3-icon bp3-icon-endorsed"
                      icon="endorsed"
                    >
                      <svg
                        data-icon="endorsed"
                        fill="#0F9960"
                        height={35}
                        viewBox="0 0 20 20"
                        width={35}
                      >
                        <desc>
                          endorsed
                        </desc>
                        <path
                          d="M19.83 9.38L18.81 7.6V5.62c0-.45-.23-.85-.61-1.08l-1.71-1-1.02-1.76a1.25 1.25 0 00-1.08-.61h-2.03l-1.74-1c-.38-.23-.87-.23-1.25 0l-1.74 1H5.65c-.44 0-.85.23-1.08.61L3.58 3.5l-1.8 1.04c-.38.24-.62.64-.62 1.08v2.06L.17 9.4c-.11.19-.17.4-.17.61s.06.42.17.61l.99 1.72v2.06c0 .45.23.85.61 1.08l1.78 1.02.99 1.72c.23.38.63.61 1.08.61h1.99l1.74 1c.19.11.41.17.62.17.21 0 .42-.06.61-.17l1.74-1h2.03c.44 0 .85-.23 1.08-.61l1.02-1.76 1.71-1c.38-.23.61-.64.61-1.08v-1.97l1.02-1.78c.27-.38.27-.85.04-1.25zm-5.08-.71l-5.01 5.01c-.18.18-.43.29-.71.29-.28 0-.53-.11-.71-.29l-3.01-3.01a1.003 1.003 0 011.42-1.42l2.3 2.3 4.31-4.3a1.003 1.003 0 011.71.71c0 .28-.12.53-.3.71z"
                          fillRule="evenodd"
                          key="0"
                        />
                      </svg>
                    </span>
                  </Blueprint3.Icon>
                  <Component>
                    <h5
                      className="bp3-heading"
                    />
                  </Component>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            center="xs"
          >
            <div
              className="row center-xs"
            >
              <Col
                xs={10}
              >
                <div
                  className="col-xs-10"
                >
                  <div
                    style={
                      Object {
                        "marginBottom": "10px",
                      }
                    }
                  >
                    <Blueprint3.Text
                      ellipsize={false}
                      tagName="div"
                    >
                      <div
                        className=""
                      />
                    </Blueprint3.Text>
                  </div>
                  <div
                    id="connected-devices-list-text-success"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "marginBottom": "25px",
                        "textDecoration": "underline dotted",
                      }
                    }
                  >
                    <Blueprint3.Text
                      className=""
                      ellipsize={false}
                      tagName="div"
                    >
                      <div
                        className=""
                      />
                    </Blueprint3.Text>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
          <Blueprint3.Button
            icon="repeat"
            intent="primary"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "100px",
              }
            }
          >
            <button
              className="bp3-button bp3-intent-primary"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "borderRadius": "100px",
                }
              }
              type="button"
            >
              <Blueprint3.Icon
                icon="repeat"
                key="leftIcon"
              >
                <span
                  className="bp3-icon bp3-icon-repeat"
                  icon="repeat"
                >
                  <svg
                    data-icon="repeat"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <desc>
                      repeat
                    </desc>
                    <path
                      d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                      fillRule="evenodd"
                      key="0"
                    />
                  </svg>
                </span>
              </Blueprint3.Icon>
              <Blueprint3.Icon
                key="rightIcon"
              />
            </button>
          </Blueprint3.Button>
        </div>
      </Col>
    </SuccessStep>
  </Router>
</BrowserRouter>
`;
